src/app/core/relations/relations.js

```diff
@@ -111,32 +181,58 @@ proto.removeRelation = function(relation) {
   }
 };
 
+/**
+ * @param {string} fatherId relation identifier
+ * @returns {boolean}
+ */
 proto.hasChildren = function(childId) {
   const children = this.getChildren(childId);
-  return  children ? !!children.length: false;
+  return children ? !!children.length : false;
 };
 
+/**
+ * @param {string} fatherId relation identifier
+ * @returns {boolean}
+ */
 proto.hasFathers = function(fatherId) {
   const fathers = this.getFathers(fatherId);
   return fathers ? !!fathers.length : false;
 };
 
-// get children based on father id
+/**
+ * Get children based on father id
+ * 
+ * @param {string} fatherId relation identifier
+ * @returns {object} children
+ */
 proto.getChildren = function(fatherId) {
   if (!this.isFather(fatherId)) return null;
   return this._relationsInfo.fathers[fatherId];
```

Comment:
Here the code is not very clear (there seems to be a mistake), why calling the following function:

```js
Relations::getChildren(fatherId)
```

is the father returned?

```js
return this._relationsInfo.fathers[fatherId];
```

Reply:
@Raruto this._relationsInfo.fathers store layers that are father in relations and store layerId children